### DISCUSSION TITLE
v0.0.6 - UI, subprocesses, dependencies, and more!

### DISCUSSION THREAD

**Discussion by carson-katri** [carson-katri/dream-textures#158] [Announcements]
Dependencies are now packaged with the addon, making installation much simpler! Just download the correct file for your platform below, and install it in Blender.

A [Dream Textures Discord server](https://discord.gg/EmDJ8CaWZ7) has also been created.

# Windows Installation
If you had a previous version of the addon, click `Remove` in Blender to delete it before installing the update.

1. Download the file `dream_textures-win32-cuda.7z`
2. Due to file size limits on GitHub, the addon `.zip` file is contained in the 7-Zip archive. Download [7-Zip](https://www.7-zip.org/) if you don't have it already.
3. Right click the file and select `7-Zip` > `Extract Here`
4. Open Blender > Preferences > Add-ons > Install..., then select the file `dream_textures-win32-cuda.zip` extracted previously. The addon may take a minute to be copied into Blender.
5. Follow the instructions in the addon preferences window to setup the model weights.

> If you want to remove the old dependencies to free up space on your computer, navigate to `C:\Program Files\Blender Foundation\Blender 3.3\3.3\python\lib\site-packages` and delete the packages previous packages. You can also delete the full `Blender 3.3` folder and reinstall Blender.

# macOS Installation
1. Download the file `dream_textures-darwin-arm64.zip`
2. Open Blender > Preferences > Add-ons > Install..., then select the zip file. It may take a minute to be copied into Blender.
3. Follow the instructions in the addon preferences window to setup the model weights.

# Other Platforms Installation
Other platforms are not officially supported by this addon. However, you can install it manually by following the steps under [Contributing](https://github.com/carson-katri/dream-textures#contributing) in the README.

## UI Changes
Dream Textures has moved to the side panel of the Image Editor and Shader Editor:
<img width="306" alt="Screen Shot 2022-09-27 at 10 18 49 AM" src="https://user-images.githubusercontent.com/13581484/192554426-c4fb8797-22f1-46e8-af07-170b4723a837.png">

## What's Changed
* Dream Texture: UI Changes by @blastframe in https://github.com/carson-katri/dream-textures/pull/39
* Add simple progress indicator by @carson-katri in https://github.com/carson-katri/dream-textures/pull/57
* Improved seed selector by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/69
* Fixed AttributeError with NoneType in context.area.regions by @blastframe in https://github.com/carson-katri/dream-textures/pull/74
* Generator subprocess by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/96
* Remove dependency installation requirement by @carson-katri in https://github.com/carson-katri/dream-textures/pull/98
* History and Troubleshooting Panels by @carson-katri in https://github.com/carson-katri/dream-textures/pull/132
* Direct users who downloaded source to download release by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/134
* Add preload models step by @carson-katri in https://github.com/carson-katri/dream-textures/pull/150

## New Contributors
* @blastframe made their first contribution in https://github.com/carson-katri/dream-textures/pull/39
* @NullSenseStudio made their first contribution in https://github.com/carson-katri/dream-textures/pull/69

**Full Changelog**: https://github.com/carson-katri/dream-textures/compare/0.0.5...0.0.6

<hr /><em>This discussion was created from the release <a href='https://github.com/carson-katri/dream-textures/releases/tag/0.0.6'>v0.0.6</a>.</em>

**Comment by OmidGhotbi** [comment]
last version works perfect but new version show some error and not work at all.
Here is the error message screenshot.
![image](https://user-images.githubusercontent.com/164653/192773481-c22b35cc-f9b8-47c8-97d7-8c1aa739bf25.png)

**Reply by carson-katri** [reply]
Did you remove the old version first? Also, one user had to run blender as an administrator once with the new version.

**Reply by OmidGhotbi** [reply]
Yes, of course, I also check the script and addons folder to be sure there is no remaining file. it will not happen to blender 3.3 but all other versions I test it has the same issue. maybe something is wrong with my PC but the old version works perfectly and the new version works just with blender 3.3 for me.
just to mention I'm using windows 11 64 bit

**Reply by jb0gie** [reply]
yeah i got the same thing, w/o completely removing the old package first 😅 tried that and the same. soooooo i did a fresh install of blender, ran as admin and voila Dream Textures updated 🎉 setting ma ⚖️s now

also running win11 64bit {insider preview}

**Comment by jesontotti** [comment]
Hi, when i try to generate any type of image i have this error

Process closed unexpectedly
![image](https://user-images.githubusercontent.com/75133438/192829447-07f1094f-0796-4809-b3f3-ce1b01ed1ad5.png)

i have 32 gb of ram and a geforce gtx 1070.

What can be to give me this error?

**Comment by ZabolotskiyD** [comment]
Hello! Have a problem. No one version does't work. 
PC: i7 11gen, ssd, ram 32gb , gtx 1650 4 gb, blender 3.3

Last version have next error after press Generate (Run as admin, old version deleted, blender fresh install). 

> [Read prefs: C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\config\userpref.blend
> Error: Traceback (most recent call last):
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\stable_diffusion\ldm\generate.py", line 344, in prompt2image
>     results = generator.generate(
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\stable_diffusion\ldm\dream\generator\base.py", line 70, in generate
>     image = make_image(x_T)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\autograd\grad_mode.py", line 27, in decorate_context
>     return func(*args, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\stable_diffusion\ldm\dream\generator\txt2img.py", line 30, in make_image
>     samples, _ = sampler.sample(
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\autograd\grad_mode.py", line 27, in decorate_context
>     return func(*args, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\stable_diffusion\ldm\models\diffusion\ksampler.py", line 83, in sample
>     K.sampling.__dict__[f'sample_{self.schedule}'](
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\autograd\grad_mode.py", line 27, in decorate_context
>     return func(*args, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\k_diffusion\sampling.py", line 186, in sample_lms
>     denoised = model(x, sigmas[i] * s_in, **extra_args)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\nn\modules\module.py", line 1110, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\stable_diffusion\ldm\models\diffusion\ksampler.py", line 16, in forward
>     uncond, cond = self.inner_model(x_in, sigma_in, cond=cond_in).chunk(2)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\nn\modules\module.py", line 1110, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\k_diffusion\external.py", line 100, in forward
>     eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\k_diffusion\external.py", line 126, in get_eps
>     return self.inner_model.apply_model(*args, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\stable_diffusion\ldm\models\diffusion\ddpm.py", line 1440, in apply_model
>     x_recon = self.model(x_noisy, t, **cond)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\nn\modules\module.py", line 1110, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\stable_diffusion\ldm\models\diffusion\ddpm.py", line 2148, in forward
>     out = self.diffusion_model(x, t, context=cc)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\nn\modules\module.py", line 1110, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\stable_diffusion\ldm\modules\diffusionmodules\openaimodel.py", line 798, in forward
>     emb = self.time_embed(t_emb)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\nn\modules\module.py", line 1110, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\nn\modules\container.py", line 141, in forward
>     input = module(input)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\nn\modules\module.py", line 1110, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "C:\Users\Lefor\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\torch\nn\modules\linear.py", line 103, in forward
>     return F.linear(input, self.weight, self.bias)
> RuntimeError: expected scalar type Half but found Float

**Reply by carson-katri** [reply]
Trying going to the folder `~/.caché` (in finder press Shift+CMD+G and paste in that path). Then delete the folders `huggingface` and `torch` and run it again.

**Reply by carson-katri** [reply]
@ZabolotskiyD for this you need to enable full precision under “advanced configuration”

**Reply by ZabolotskiyD** [reply]
@carson-katri Great thanks! Error is gone, but error is come. 
It's problem of my GPU (gtx1650 4GB) ?  In Configeration I set 64x64 possible minimum.

> Error: RuntimeError('CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 4.00 GiB total capacity; 3.41 GiB already allocated; 0 bytes free; 3.46 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF')

**Comment by nsongha** [comment]
Hi, I'm trying to run this on Blender 3.3 Mac Monterey but it keeps showing this error.
Can anyone help me to fix this?
![193236660-d7358f8d-cc22-4e51-a5b2-b39b49c9fe0e](https://user-images.githubusercontent.com/64000580/193294735-f28c55e1-0dea-43e5-ad26-d677e4ccaf81.png)

**Reply by hboteler** [reply]
I have same problem and message on my MacBook Pro (OS Monterey/Blender 3.3 & 3.4 alpha). Have you found a solution yet?

**Reply by nsongha** [reply]
Not yet, I'm trying old versions but still no luck. I'm on M1 by the way.

**Comment by qfb594** [comment]
Why can't the model download, prompt to authorize?

**Reply by carson-katri** [reply]
I’m not sure I understand

**Comment by qfb594** [comment]
![QQ浏览器截图20221001105603](https://user-images.githubusercontent.com/48076384/193383827-79b734a9-71d3-4ffe-ba2a-20285163f49e.png)
blender Version 3.3, why is there no plugin display in the panel?

**Reply by carson-katri** [reply]
It’s in the image editor and shader editor.

**Comment by qfb594** [comment]


**Reply by carson-katri** [reply]
You need to enable “Full Precision” under “Advanced configuration”

**Comment by Velgardi** [comment]


**Reply by qfb594** [reply]
你起码要用大内存的显卡，小显卡就不要玩这个了。

**Reply by ZabolotskiyD** [reply]
@qfb594   how much memory is enought?

**Reply by Velgardi** [reply]
@qfb594 And with a small amount of memory it is impossible to implement in any way?

**Reply by qfb594** [reply]
我试过，不行，建议用2060 以上的显卡尝试。

**Comment by caith-h** [comment]
The Sampler isn't working for me anymore.
To be more exact, it pretends to work - but is actually using KLMS no matter what I select. In the History, the Sampler I selected is shown - but in actuality, it's still using KLMS, which can be tested by running the same seed with different Samplers, and only getting back the KLMS 'output'.

**Reply by caith-h** [reply]
FYI, I've tried this on two different machines. One of which hadn't had the addon installed at all before this update.

**Comment by Boboknight11** [comment]
I love this addon and am so happy the random seed integer overflow was fix. But please bring back input field for the concept artist. I used that SO much and I keep track the exact generation parameters when I find a good image so I can go back and recreate the image for further modification, but now I can't recreate my images anymore even if I try to put the concept artist in the prompt.

**Reply by JakubScorch** [reply]
I would love that too. Promting the style/artist seems to give unpredictable results.
